src/control/server/storage/bdev/backend.go

```diff
@@ -114,33 +113,222 @@ func defaultBackend(log logging.Logger) *spdkBackend {
+}
+
+// Reset will perform a lookup on the requested target user to validate existence
+// then reset non-VMD NVMe devices for use by the OS/kernel.
+// If EnableVmd is true in request then attempt to use VMD NVMe devices.
+// If DisableCleanHugePages is false in request then cleanup any leftover hugepages
+// owned by the target user.
+// Backend call executes the SPDK setup.sh script to rebind PCI devices as selected by
+// bdev_include and bdev_exclude list filters provided in the server config file.
+func (sb *spdkBackend) Reset(req storage.BdevPrepareRequest) error {
+	sb.log.Debugf("spdk backend reset (script call): %+v", req)
+	return sb.reset(req, detectVMD)
+}
+
+// Prepare will perform a lookup on the requested target user to validate existence
+// then prepare non-VMD NVMe devices for use with SPDK.
+// If EnableVmd is true in request then attempt to use VMD NVMe devices.
+// If DisableCleanHugePages is false in request then cleanup any leftover hugepages
+// owned by the target user.
+// Backend call executes the SPDK setup.sh script to rebind PCI devices as selected by
+// bdev_include and bdev_exclude list filters provided in the server config file.
+func (sb *spdkBackend) Prepare(req storage.BdevPrepareRequest) (*storage.BdevPrepareResponse, error) {
+	sb.log.Debugf("spdk backend prepare (script call): %+v", req)
+	return sb.prepare(req, user.Lookup, detectVMD, cleanHugePages)
+}
+
+// canAccessBdevs evaluates if any specified Bdevs are not accessible.
+//
+// Specified Bdevs can be VMD addresses.
+//
+// Return any device addresses missing from the discovered bdevs and ok set to true
+// if no devices are missing.
+func canAccessBdevs(cfgBdevs []string, discoveredBdevs storage.NvmeControllers) ([]string, bool) {
+	var missing []string
+
+	for _, pciAddr := range cfgBdevs {
+		var found bool
+
+		for _, ctrlr := range discoveredBdevs {
+			if ctrlr.PciAddr == pciAddr {
+				found = true
+				break
+			}
+		}
+		if !found {
+			missing = append(missing, pciAddr)
+		}
+	}
+
+	return missing, len(missing) == 0
+}
+
+// checkCfgBdevsExist performs validation on NVMe returned from initial scan.
+func checkCfgBdevsExist(log logging.Logger, bdevs storage.NvmeControllers, engineStorage map[uint32]*storage.Config, vmdEnabled bool) error {
+	if len(engineStorage) == 0 {
+		return nil
+	}
+
+	for idx, storageCfg := range engineStorage {
+		for _, tierCfg := range storageCfg.Tiers {
+			if !tierCfg.IsBdev() || len(tierCfg.Bdev.DeviceList) == 0 {
+				continue
+			}
+			cfgBdevs := tierCfg.Bdev.DeviceList
+
+			// If VMD devices have been prepared, enabled will be set in the scanRequest by
+			// the BdevAdminForwarder.
+			if vmdEnabled {
+				msg := fmt.Sprintf("vmd detected, checking addresses (input %v, existing %v)",
+					cfgBdevs, bdevs)
+
+				dl, err := substVMDAddrs(cfgBdevs, bdevs)
+				if err != nil {
+					return err
+				}
+				log.Debugf("check bdev cfg for engine-%d: %s: new %s", idx, msg, dl)
```

Comment:
this output is a little confusing:
DEBUG 00:44:16.196015 backend.go:285: check bdev cfg for engine-0: vmd detected, checking addresses (input [0000:5d:05.5], existing 5d0505:01:00.0, 5d0505:03:00.0): new [5d0505:01:00.0 5d0505:03:00.0]
maybe "new" should be "added bdevs" or something like that?